### PR TITLE
add locker

### PR DESCRIPTION
**Pull request by jiyeyuran**

### CODE REVIEW

**Comment by asim** [comment]
It's a mock registry. Why would you do this?

**Comment by jiyeyuran** [comment]
I want to test multi micro services on one process. My company also wants to package all micro services into one binary application to simplify the deployment.

File: registry/mock/mock.go [hunk 1]
```diff
@@ -1,11 +1,13 @@
-// Package mock provides a mock registry for testing
```
[the remark this answers is not in the document]
Comment (jiyeyuran) [reply registry/mock/mock.go]: Added it back. Does go-micro need a mock server?

[the remark this answers is not in the document]
Comment (jiyeyuran) [reply registry/mock/mock.go]: Mock server won't need to listen port, but rpc server needs.

**Pull request merged**